src/main/scala/beam/router/RoutingWorker.scala

```diff
@@ -107,8 +112,23 @@ class RoutingWorker(workerParams: R5Parameters) extends Actor with ActorLogging
     if (carRouter == "staticGH" || carRouter == "quasiDynamicGH") {
       new Directory(new File(graphHopperDir)).deleteRecursively()
       createWalkGraphHopper()
-      createCarGraphHoppers()
+      createCarGraphHoppers(carRouter)
       askForMoreWork()
+    } else if (carRouter == "hybridGH") {
+      createWalkGraphHopper()
```

Comment:
Should these methods return the object itself? Then you don't have to do any of the casting below..which is not compile safe.

Reply:
Probably you are right. I'll refactor these methods.

Reply:
I cannot change createCarGraphHoppers to return the object because it also does some side effects. I resolved this issue by assigning the right subtypes to the fields.